### PR TITLE
Annotate Azure.Storage.Blobs for trim/AOT compat

### PR DESCRIPTION
**Pull request by sbomer**
Adds trim/AOT annotations to Azure.Storage.Blobs.

Opts the project into IsTrimmable/IsAotCompatible, and the AOT test app ci check.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.net6.0.cs [hunk 1]
```diff
@@ -1998,6 +1998,7 @@ public static partial class BlobClientBuilderExtensions
...
         public static Azure.Core.Extensions.IAzureClientBuilder<Azure.Storage.Blobs.BlobServiceClient, Azure.Storage.Blobs.BlobClientOptions> AddBlobServiceClient<TBuilder>(this TBuilder builder, System.Uri serviceUri) where TBuilder : Azure.Core.Extensions.IAzureClientFactoryBuilderWithCredential { throw null; }
         public static Azure.Core.Extensions.IAzureClientBuilder<Azure.Storage.Blobs.BlobServiceClient, Azure.Storage.Blobs.BlobClientOptions> AddBlobServiceClient<TBuilder>(this TBuilder builder, System.Uri serviceUri, Azure.Storage.StorageSharedKeyCredential sharedKeyCredential) where TBuilder : Azure.Core.Extensions.IAzureClientFactoryBuilder { throw null; }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Binding strongly typed objects to configuration values is not supported with trimming. Use the Configuration Binder Source Generator (EnableConfigurationBindingGenerator=true) instead.")]
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.net6.0.cs:2001]: This file is actually generated by our eng/ExportsAPI.ps1 script.

This change is fine, I just wonder if there's a way to have this be generated every single time that way this doesn't get erased
@seanmcc-msft @jsquire @JoshLove-msft Could anyone chime in on this? I might not know the right person for this.

File: sdk/storage/ci.yml [hunk 2]
```diff
@@ -71,3 +71,7 @@ extends:
       skipSmokeTests: true
     TestSetupSteps:
     - template: /sdk/storage/tests-install-azurite.yml
+    CheckAOTCompat: true
+    AOTTestInputs:
+    - ArtifactName: Azure.Storage.Blobs
```
Comment (amnguye) [inline sdk/storage/ci.yml:76]: I wonder if we should be doing this for the rest of our packages.

File: sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.net6.0.cs (hunk 1 above)
Comment (jsquire) [reply sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.net6.0.cs:2001]: This is just the API listing file used for verifying that we don't make breaking changes between versions.    There shouldn't be a need to add anything here directly.

File: sdk/storage/ci.yml (hunk 2 above)
Comment (amnguye) [reply sdk/storage/ci.yml:76]: https://github.com/Azure/azure-sdk-for-net/issues/46906

**Review by amnguye — APPROVED** [review]

**Pull request merged**